src/ParseServer.js

```diff
@@ -184,6 +184,18 @@ class ParseServer {
     api.use(middlewares.allowMethodOverride);
     api.use(middlewares.handleParseHeaders);
 
+    if (options.middleware) {
+      let middleware;
+      if (typeof options.middleware == 'string') {
+        middleware = require(path.resolve(process.cwd(), options.middleware));
+      } else if (typeof options.middleware == 'function') {
+        middleware = options.middleware; // use as-is let express fail
+      } else {
+        throw "argument 'middleware' must either be a string or a function";
+      }
+      api.use(middleware);
+    }
+
```

Comment:
```js
it('can create server with middleware', async () => {
    const obj = {
      middleware: function (req, res, next) {
        next();
      },
    };
    const spy = spyOn(obj, 'middleware').and.callThrough();
    const parseServer = await new Promise((resolve, reject) => {
      const server = ParseServer.ParseServer({
        ...defaultConfiguration,
        ...{
          appId: 'anOtherTestApp',
          masterKey: 'anOtherTestMasterKey',
          serverURL: 'http://localhost:12667/parse',
          serverStartComplete(error) {
            if (error) {
              reject(error);
            } else {
              resolve(server);
            }
          },
          middleware: obj.middleware,
        },
      });
    });
    const app = express();
    app.use('/parse', parseServer);
    await new Promise(resolve => app.listen(12667, resolve));
    expect(Parse.applicationId).toEqual('anOtherTestApp');
    expect(Parse.serverURL).toEqual('http://localhost:12667/parse');
    await new Parse.Query('AnObject').find();
    expect(spy).toHaveBeenCalled();
  });
```

Can you please add this to [index.spec.js](https://github.com/parse-community/parse-server/blob/alpha/spec/index.spec.js#L509)

Reply:
done!
thanks for the review.

Reply:
sorry, you'll also need to close the server after the test finishes

replace

`await new Promise(resolve => app.listen(12667, resolve));` with `const httpServer = app.listen(12667)`

then call `httpServer.close()` at the end of the test